src/SFA.DAS.Reservations.Web.UnitTests/Models/WhenConstructingAReservationViewModel.cs

```diff
@@ -87,14 +93,28 @@ public class WhenConstructingAReservationViewModel
         }
 
         [Test, AutoData]
-        public void Then_If_The_ApprenticeUrl_Is_Available_The_Url_Is_Created_If_Supplied(
+        public void And_Has_Ukprn_And_ApprenticeUrl_Then_Provider_Url_Is_Created(
             Reservation reservation,
             string url,
+            string accountHashedId,
             string alephid)
         {
-            var viewModel = new ReservationViewModel(reservation, url, alephid);
+            var viewModel = new ReservationViewModel(reservation, url, accountHashedId, alephid);
 
             Assert.AreEqual($"{url}/{reservation.ProviderId}/unapproved/add-apprentice?reservationId={reservation.Id}&employerAccountLegalEntityPublicHashedId={alephid}&startMonthYear={reservation.StartDate:MMyyyy}&courseCode={reservation.Course.Id}", viewModel.ApprenticeUrl);
         }
+
+        [Test, AutoData]
+        public void And_No_Ukprn_And_Has_ApprenticeUrl_Then_Employer_Url_Is_Created(
+            Reservation reservation,
+            string url,
+            string accountHashedId,
+            string alephid)
+        {
+            reservation.ProviderId = null;
+            var viewModel = new ReservationViewModel(reservation, url, accountHashedId, alephid);
```

Comment:
small gripe, try not to over abbreviate variable names even in tests as someone who is looking at this might not clock what alephid stands for. Changing it to alePubHashId which isn't much of a change at least gives the reader some idea that its a hash of some sort.